src/pages/ShowResource.tsx

```diff
@@ -18,57 +20,103 @@ function ShowResource({ location }: RouteComponentProps) {
-  ) {
-    return <Swap swap={entity} reload={reload} />;
-  } else if (error) {
-    return (
-      <React.Fragment>
-        <Typography variant="h3" align="center" data-cy="404-typography">
-          404 Resource Not Found
-        </Typography>
-        {error && error.message === "Network Error" && (
+  const [showLoading, setShowLoading] = useState(false);
+  const [allowReload, setAllowReload] = useState(true);
+
+  useEffect(() => {
+    if (isLoading) {
+      setShowLoading(true);
+    }
+  }, [isLoading]);
+
+  useInterval(() => reload(), allowReload && !isLoading ? 15000 : null);
+
+  function resource() {
+    if (
+      !axiosError &&
+      entity &&
+      entity.class &&
+      entity.class.includes("swaps")
+    ) {
+      return (
+        <SwapList
+          swaps={entity.entities as EmbeddedRepresentationSubEntity[]}
+          reload={reload}
+          setAllowReload={setAllowReload}
+        />
+      );
+    } else if (
+      !axiosError &&
+      entity &&
+      entity.class &&
+      entity.class.includes("swap")
+    ) {
+      return (
+        <Swap swap={entity} reload={reload} setAllowReload={setAllowReload} />
+      );
+    } else if (axiosError) {
+      if (
+        axiosError.response &&
+        axiosError.response.status &&
+        Math.floor(axiosError.response.status / 100) !== 2
+      ) {
+        return (
+          <Typography variant="h3" align="center" data-cy="404-typography">
+            404 Resource Not Found
+          </Typography>
+        );
+      } else {
+        // Network error
+        return (
+          <React.Fragment>
+            <Typography variant="h3" align="center" data-cy="404-typography">
+              404 Resource Not Found
+            </Typography>
+            <ErrorSnackbar
+              message="Failed to fetch resource. Is your COMIT node running?"
+              open={true}
+            />
+          </React.Fragment>
+        );
+      }
+    } else if (!entity) {
+      return;
+    } else {
+      return (
+        <React.Fragment>
+          <Typography variant="h3" align="center" data-cy="bad-json-typography">
+            Bad JSON
+          </Typography>
           <ErrorSnackbar
-            message="Failed to fetch resource. Is your COMIT node running?"
+            message="Could not handle comit node's response. Are your comit-i and comit node versions compatible?"
             open={true}
           />
-        )}
-      </React.Fragment>
-    );
-  } else {
-    return (
-      <React.Fragment>
-        <Typography variant="h3" align="center" data-cy="bad-json-typography">
-          Bad JSON
-        </Typography>
-        <ErrorSnackbar
-          message="Could not handle comit node's response. Are your comit-i and comit node versions compatible?"
-          open={true}
-        />
-      </React.Fragment>
-    );
+        </React.Fragment>
+      );
+    }
   }
+
+  return (
+    <React.Fragment>
+      {resource()}
```

Comment:
Calling a function that returns a component is not very idiomatic react because you are loosing out on some the performance optimizations React makes when diffing the tree.
As soon as a function returns `JSX`, it is a component by itself and you can call it through `JSX` (but it has to have a `PascalCase` name):

```tsx
<Resource />
```

Reply:
Thank you. I suspected it was incorrect but it worked.

Reply:
Trying to fix this caused a weird bug: action dialogs never open now.

Reply:
I think you can't make a component that closes over local variabes. Changing those will not trigger a re-render. So either stick with the function or introduce props.